src/agent/holiday_hours.rs

```diff
@@ -0,0 +1,376 @@
+                };
+
+                if start < end {
+                    Ok(HolidayDayKind::TimeRange(start, end))
+                } else {
+                    Err(anyhow!("Incorrect time range: start must come before end"))
+                }
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::*,
+        chrono::NaiveDateTime,
+    };
+
+    #[test]
+    fn test_parsing_single_day() -> Result<()> {
+        let s = "0422/0900-1700";
+
+        let parsed: HolidaySchedule = s.parse()?;
+
+        let expected = HolidaySchedule {
+            timezone: None,
+            days:     vec![HolidayDaySchedule {
+                month: 4,
+                day:   22,
+                kind:  HolidayDayKind::TimeRange(
+                    NaiveTime::from_hms_opt(9, 0, 0).unwrap(),
+                    NaiveTime::from_hms_opt(17, 0, 0).unwrap(),
+                ),
+            }],
+        };
+
+        assert_eq!(parsed, expected);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_parsing_multiple_days() -> Result<()> {
+        let s = "0422/0900-1700,1109/0930-1730,1201/O,1225/C,1231/0900-1700";
+
+        let parsed: HolidaySchedule = s.parse()?;
+
+        let expected = HolidaySchedule {
+            timezone: None,
+            days:     vec![
+                HolidayDaySchedule {
+                    month: 4,
+                    day:   22,
+                    kind:  HolidayDayKind::TimeRange(
+                        NaiveTime::from_hms_opt(9, 0, 0).unwrap(),
+                        NaiveTime::from_hms_opt(17, 0, 0).unwrap(),
+                    ),
+                },
+                HolidayDaySchedule {
+                    month: 11,
+                    day:   9,
+                    kind:  HolidayDayKind::TimeRange(
+                        NaiveTime::from_hms_opt(9, 30, 0).unwrap(),
+                        NaiveTime::from_hms_opt(17, 30, 0).unwrap(),
+                    ),
+                },
+                HolidayDaySchedule {
+                    month: 12,
+                    day:   1,
+                    kind:  HolidayDayKind::Open,
+                },
+                HolidayDaySchedule {
+                    month: 12,
+                    day:   25,
+                    kind:  HolidayDayKind::Closed,
+                },
+                HolidayDaySchedule {
+                    month: 12,
+                    day:   31,
+                    kind:  HolidayDayKind::TimeRange(
+                        NaiveTime::from_hms_opt(9, 0, 0).unwrap(),
+                        NaiveTime::from_hms_opt(17, 0, 0).unwrap(),
+                    ),
+                },
+            ],
+        };
+
+        assert_eq!(parsed, expected);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_parsing_month_without_leading_zero_is_error() {
+        let s = "422/0900-1700";
+
+        let parsing_result: Result<HolidaySchedule> = s.parse();
+        assert!(parsing_result.is_err());
+    }
```

Comment:
I would suggest including the `proptest` and `arbitrary` crate's and adding a test that uses this to properly try and detect any parser failures, the test would look something like this after you define the arbitrary impls:

```
    proptest! {
        #[test]
        fn test_parse_schedule(holiday_schedule in holiday_schedule_strategy()) {
            let schedule_string = holiday_schedule.to_string();
            let parsed_schedule = schedule_string.parse::<HolidaySchedule>();
            prop_assert!(parsed_schedule.is_ok());
            prop_assert_eq!(parsed_schedule.unwrap(), holiday_schedule);
        }
    }
```

Reply:
Done